### PR TITLE
fix(manager/gradle): drop stdout from gradlew :dependency command

### PR DESCRIPTION
**Pull request by ainame**
## Changes

This PR is for Gradle manager module. I added `> /dev/null` to the command used for updating and writing `gradle.lockfile`. This way `./gradlew :dependencies --update-locks MODULE_NAMES` won't get `stdout maxBuffer exceeded` error which is caused by floods of text output from `./gradlew :dependencies` that renders the dependency graphs, especially at a scale app adopting modularisation architecture. 

While `stdout` from a command is important for visibility while debugging in general, in this case, Gradle outputs irrelevant information. As long as we secure `stderr` still, it's safer to drop.  

## Context

The background is explained on this discussion page https://github.com/renovatebot/renovate/discussions/20736

In short, although Gradle's official document recommends using `./gradlew :dependencies --write/--update-locks`, `:dependencies` command is meant to visualise the dependency graphs on the console by rendering them on console with text, which we can expect the command to produce a huge amount of irrelevant output to `stdout`. With `--wirte/update-locks` option, you can reflect the state of dependency resolving during the command.

https://github.com/gradle/gradle/blob/master/subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/viewing_debugging_dependencies.adoc

---

The same issue was reported in the past https://github.com/renovatebot/renovate/issues/4971, and Renovate changed the default `maxBuffer` value to 10MB, but our project now produces 59MB of text data rendering dependency graphs. That implies changing `maxBuffer` isn't a sustainable approach for large apps.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — APPROVED** [review]

**Comment by secustor** [comment]
This will break usage of Renovate on Windows.

File: lib/modules/manager/gradle/artifacts.ts [hunk 1]
```diff
@@ -157,6 +157,11 @@ export async function updateArtifacts({
       cmd += ` --update-locks ${updatedDepNames.map(quote).join(',')}`;
     }
 
+    // `./gradlew :dependencies` command can output huge text due to `:dependencies`
+    // that renders dependency graphs. Given the output can exceed `ExecOptions.maxBuffer` size,
+    // drop stdout from the command.
+    cmd += ' > /dev/null';
```
Comment (viceice) [inline lib/modules/manager/gradle/artifacts.ts:163]: would Gradle log errors to stderr?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs consideration of Windows so that it would only happen when binarySource=docker in such cases

File: lib/modules/manager/gradle/artifacts.ts (hunk 1 above)
Comment (ainame) [reply lib/modules/manager/gradle/artifacts.ts:163]: Yes, Gradle still leaves error messages to `stderr`.

**Comment by ainame** [comment]
@rarkins @secustor 

For now, I excluded Windows support (without docker), since I didn't have a clear idea of how to properly support various Windows environments; like PowerShell, Command Prompt, Cygwin, etc... Let me know if there's a better way to do this.

File: lib/modules/manager/gradle-wrapper/utils.ts [hunk 2]
```diff
@@ -21,6 +21,17 @@ export function gradleWrapperFileName(): string {
   return './gradlew';
 }
 
+export function nullRedirectionCommand(): string {
+  if (
+    os.platform() === 'win32' &&
+    GlobalConfig.get('binarySource') !== 'docker'
+  ) {
+    // TODO: Windows environment without docker needs to be implemented
```
Comment (secustor) [inline lib/modules/manager/gradle-wrapper/utils.ts:29]: Add a debug log message here, else we will not know what is happening.

**Review by secustor — CHANGES_REQUESTED** [review]

File: lib/modules/manager/gradle/readme.md [hunk 3]
```diff
@@ -1,2 +1,6 @@
 The `gradle` manager uses a custom parser written in JavaScript, similar to many others managers.
 It does not call `gradle` directly in order to extract a list of dependencies.
+
+### Updating lockfiles
+
+Updating lockfiles is done with `./gradlew :dependencies --wirte/update-locks` command. This command can output excessive text to the console. While running the command, the output to stdout is dropped when you run Renovate on most platfroms other than Windows.
```
Comment (secustor) [inline lib/modules/manager/gradle/readme.md:6]: Use one line per sentence, please. 
See our developer docs for more infos.

**Review by secustor — CHANGES_REQUESTED** [review]

File: lib/modules/manager/gradle/readme.md (hunk 3 above)
Comment (ainame) [reply lib/modules/manager/gradle/readme.md]: Thanks. TIL about one line per sentence. I ran `yarn doc-fix` but didn't get updates🤔 I manually fixed this.

**Review by secustor — DISMISSED** [review]

File: lib/modules/manager/gradle/readme.md [hunk 4]
```diff
@@ -1,2 +1,8 @@
 The `gradle` manager uses a custom parser written in JavaScript, similar to many others managers.
 It does not call `gradle` directly in order to extract a list of dependencies.
+
+### Updating lockfiles
+
+Updating lockfiles is done with `./gradlew :dependencies --wirte/update-locks` command.
+This command can output excessive text to the console.
+While running the command, the output to stdout is dropped when you run Renovate on most platfroms other than Windows.
```
Comment (rarkins) [inline lib/modules/manager/gradle/readme.md:8]:
```suggestion
While running the command, the output to stdout is dropped when you run Renovate on most platforms other than Windows.
```

**Review by rarkins — APPROVED** [review]

**Pull request merged**